src/components/PhotoAnalysis/PhotoAnalysis.tsx

```diff
@@ -25,70 +25,95 @@ const Analysis: React.FC = () => {
   >("")
   const [imageSrc, setImageSrc] = useState<string>()
   let labels: string[] = [""]
-  let labelData: Label[]
+  let labelData: LabelType[] = []
   const [isLoading, setIsLoading] = useState(false)
+  const [isError, setError] = useState(false)
 
-  const identifyImageLabels = async (event: any) => {
-    setIsLoading(true)
-
+  const identifyImageResponse = async (event: any) => {
     const files = (event.target as HTMLInputElement).files
     const file = files![0]
-    setImageSrc(URL.createObjectURL(file))
-
-    await Predictions.identify({
-      labels: {
-        source: {
-          file,
+    if (isValidFileType(file)) {
+      setIsLoading(true)
+      setImageSrc(URL.createObjectURL(file))
+      await Predictions.identify({
+        labels: {
+          source: {
+            file,
+          },
+          type: "ALL",
         },
-        type: "ALL",
-      },
-    })
-      .then(response => {
-        setRekognitionResponse(response)
-        setIsLoading(false)
       })
-      .catch(err => setRekognitionResponse(JSON.stringify(err, null, 2)))
+        .then(response => {
+          setRekognitionResponse(response)
+          setIsLoading(false)
+        })
+        .catch(err => setRekognitionResponse(JSON.stringify(err, null, 2)))
+    } else setError(true)
   }
 
   const processRekognitionLabels = (
```

Comment:
Should this be in another file? I like having it here and I am not entirely sure how to extract it efficiently to a custom hook. Should I be investing time into this or do you think it's something we should come back to?